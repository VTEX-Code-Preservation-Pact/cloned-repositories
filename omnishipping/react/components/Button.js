import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Button extends Component {
  render() {
    const {
      kind,
      size,
      border,
      isInline,
      isLonely,
      isLoading,
      isDisabled,
      isFixedValue,
      isNotChecked,
      icon,
      title,
      moreClassName = '',
      ...rest
    } = this.props

    let style = `btn relative tc bn fw3 ${moreClassName} `

    switch (kind) {
      case 'success':
        style += 'btn-success bg-green white '
        break
      case 'primary':
        style += 'bg-blue white '
        break
      case 'info':
        style += 'bg-blue white '
        break
      case 'alert':
        style += 'bg-gold white '
        break
      case 'danger':
        style += 'bg-light-red white '
        break
      case 'muted':
        style += 'bg-white gray '
        break
      case 'link':
      case 'light-primary':
        style += 'bg-white blue '
        break
      case 'light-danger':
        style += 'bg-white light-red '
        break
      default:
        style += 'bg-near-white blue '
    }

    switch (size) {
      case 'xs':
        style += 'pa1 f7 f6-ns '
        break
      case 'sm':
        style += 'pa2 pa3-ns f6 f5-ns lh-copy '
        break
      case 'lg':
        style += 'btn-large pv3 ph4 pa4-ns f4 f3-ns '
        break
      case 'xl':
        style += 'pa4 pa5-ns f3 f2-ns '
        break
      default:
        style += 'pv3 lh-solid ph4 f5 fw5 '
    }

    switch (border) {
      case 'dotted':
        style += 'b--moon-gray b--dotted bw1 '
        break
      default:
        style += ''
    }

    style += !isInline ? 'db w-100 br3 ' : ''
    style += isDisabled ? 'o-50 ' : ''
    style += !isLonely ? 'mh1 ' : ''

    return (
      <button className={style} disabled={isDisabled} {...rest} type="button">
        {isLoading && (
          <i className="fa fa-circle-o-notch fa-spin fa-fw mr1 absolute left-1" />
        )}
        {icon && <i className={`mr1 ${icon}`} />}
        {title}
        {!isFixedValue && <i className="fa fa-caret-down absolute right-1" />}
        {!isNotChecked && <i className="fa fa-check absolute right-1" />}
      </button>
    )
  }
}

Button.defaultProps = {
  isFixedValue: true,
  isNotChecked: true,
}

Button.propTypes = {
  border: PropTypes.string,
  icon: PropTypes.string,
  isDisabled: PropTypes.bool,
  isFixedValue: PropTypes.bool,
  isInline: PropTypes.bool,
  isLoading: PropTypes.bool,
  isLonely: PropTypes.bool,
  isNotChecked: PropTypes.bool,
  kind: PropTypes.string,
  moreClassName: PropTypes.string,
  size: PropTypes.string,
  title: PropTypes.string,
}

export default Button
