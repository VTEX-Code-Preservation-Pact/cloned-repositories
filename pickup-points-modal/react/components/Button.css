/* basic style */

.btn {
  border-radius: 4px;
  font-size: 12px;
  padding: 6px 8px;
  text-transform: uppercase;
}

/* block */

.block {
  display: block;
  width: 100%;
}

/* large */

.lg {
  font-size: 14px;
  padding: 12px 16px;
}

/* primary button */

.primary {
  background: #368df7;
  border: 1px solid #368df7;
  color: #fff;
}

.primary:hover {
  background: #2c7de0;
  border-color: #2c7de0;
}

/* secondary button */

.secondary {
  background: none;
  border: 1px solid #368df7;
  color: #368df7;
}

.secondary:hover {
  border-color: #2c7de0;
  color: #2c7de0;
}
