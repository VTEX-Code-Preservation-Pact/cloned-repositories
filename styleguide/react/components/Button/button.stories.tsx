import React from 'react'
import PropTypes from 'prop-types'
import { Story, Meta } from '@storybook/react'
import { action } from '@storybook/addon-actions'

import Button, { propTypes } from '.'

type ButtonProps = PropTypes.InferProps<typeof propTypes>

const variations = [
  'primary',
  'secondary',
  'tertiary',
  'danger',
  'danger-tertiary',
  'inverted-tertiary',
]
const sizes = ['small', 'regular', 'large']

export default {
  title: 'Components/Button',
  component: Button,
  argTypes: {
    variation: {
      control: {
        type: 'select',
        options: variations,
      },
    },
    size: {
      control: {
        type: 'inline-radio',
        options: sizes,
      },
    },
    onClick: {
      action: 'button-click',
    },
  },
} as Meta

const Template: Story<ButtonProps> = args => <Button {...args} type="button" />

export const Default = Template.bind({})
Default.args = {
  variation: 'primary',
  isLoading: false,
  disabled: false,
  block: false,
  size: 'regular',
  children: 'Button',
}

export const withLoadingState = Template.bind({})
withLoadingState.args = {
  ...Default.args,
  isLoading: true,
}

export const withDifferentVariations = () => (
  <>
    {variations.map(variation => (
      <>
        <Button type="button" variation={variation}>
          {variation}
        </Button>
        <br />
        <br />
      </>
    ))}
  </>
)

export const withOnMouseEvents = Template.bind({})
withOnMouseEvents.args = {
  ...Default.args,
  onMouseDown: action('button-on-mouse-down'),
  onMouseEnter: action('button-on-mouse-enter'),
  onMouseOver: action('button-on-mouse-over'),
  onMouseLeave: action('button-on-mouse-leave'),
  onMouseOut: action('button-on-mouse-out'),
  onMouseUp: action('button-on-mouse-up'),
}

export const withDifferentSizes = () => (
  <>
    {sizes.map(size => (
      <>
        <Button type="button" size={size}>
          {size}
        </Button>
        <br />
        <br />
      </>
    ))}
  </>
)
