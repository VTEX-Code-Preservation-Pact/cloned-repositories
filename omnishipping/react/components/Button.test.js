import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('Button', () => {
  it('should render self and subcomponents', () => {
    const props = {
      title: 'title',
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.find('button').hasClass('relative')).toBe(true)

    expect(wrapper.find('button').text()).toBe('title')
  })

  it('should render success mode', () => {
    const props = {
      title: 'title',
      kind: 'success',
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render primary mode', () => {
    const props = {
      title: 'title',
      kind: 'primary',
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render info mode', () => {
    const props = {
      title: 'title',
      kind: 'info',
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render alert mode', () => {
    const props = {
      title: 'title',
      kind: 'alert',
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render danger mode', () => {
    const props = {
      title: 'title',
      kind: 'danger',
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render muted mode with size xs and is not inline', () => {
    const props = {
      title: 'title',
      kind: 'muted',
      size: 'xs',
      isInline: true,
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render link mode with size sm and disabled and is not lonely', () => {
    const props = {
      title: 'title',
      kind: 'link',
      size: 'sm',
      isDisabled: true,
      isLonely: true,
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render light-primary mode with size lg and loading', () => {
    const props = {
      title: 'title',
      kind: 'light-primary',
      size: 'lg',
      isLoading: true,
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })

  it('should render light-danger mode with size xl not checked with fixed value', () => {
    const props = {
      title: 'title',
      kind: 'light-danger',
      size: 'xl',
      isNotChecked: false,
      isFixedValue: false,
    }
    const wrapper = shallow(<Button {...props} />)

    expect(wrapper.getElements()[0]).toMatchSnapshot()
  })
})
