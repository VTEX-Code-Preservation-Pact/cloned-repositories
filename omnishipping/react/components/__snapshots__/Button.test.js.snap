// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render alert mode 1`] = `
<button
  className="btn relative tc bn fw3  bg-gold white pv3 lh-solid ph4 f5 fw5 db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  title
</button>
`;

exports[`Button should render danger mode 1`] = `
<button
  className="btn relative tc bn fw3  bg-light-red white pv3 lh-solid ph4 f5 fw5 db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  title
</button>
`;

exports[`Button should render info mode 1`] = `
<button
  className="btn relative tc bn fw3  bg-blue white pv3 lh-solid ph4 f5 fw5 db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  title
</button>
`;

exports[`Button should render light-danger mode with size xl not checked with fixed value 1`] = `
<button
  className="btn relative tc bn fw3  bg-white light-red pa4 pa5-ns f3 f2-ns db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  title
  <i
    className="fa fa-caret-down absolute right-1"
  />
  <i
    className="fa fa-check absolute right-1"
  />
</button>
`;

exports[`Button should render light-primary mode with size lg and loading 1`] = `
<button
  className="btn relative tc bn fw3  bg-white blue btn-large pv3 ph4 pa4-ns f4 f3-ns db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  <i
    className="fa fa-circle-o-notch fa-spin fa-fw mr1 absolute left-1"
  />
  title
</button>
`;

exports[`Button should render link mode with size sm and disabled and is not lonely 1`] = `
<button
  className="btn relative tc bn fw3  bg-white blue pa2 pa3-ns f6 f5-ns lh-copy db w-100 br3 o-50 "
  disabled={true}
  type="button"
>
  title
</button>
`;

exports[`Button should render muted mode with size xs and is not inline 1`] = `
<button
  className="btn relative tc bn fw3  bg-white gray pa1 f7 f6-ns mh1 "
  disabled={undefined}
  type="button"
>
  title
</button>
`;

exports[`Button should render primary mode 1`] = `
<button
  className="btn relative tc bn fw3  bg-blue white pv3 lh-solid ph4 f5 fw5 db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  title
</button>
`;

exports[`Button should render success mode 1`] = `
<button
  className="btn relative tc bn fw3  btn-success bg-green white pv3 lh-solid ph4 f5 fw5 db w-100 br3 mh1 "
  disabled={undefined}
  type="button"
>
  title
</button>
`;
